app/models/migration_upload_snapshot.rb

```diff
@@ -18,6 +18,13 @@ def mark_complete(s3_file)
     if index.present?
       files[index]["migrate_status"] = "complete"
       files[index].delete("migrate_error")
+
+      # Retrieve the checksum from AWS, as this often differs from what is migrated from DSpace
+      # Please see https://github.com/pulibrary/pdc_describe/issues/1413
+      updated = s3_file.s3_query_service.get_s3_object_attributes(key: s3_file.filename)
```

Comment:
You could pass in the returned etag instead of querying again.  That would fix the failing tests

```
   migration_snapshot.mark_complete(result[:file],result[:etag].delete('"'))
```